File: app/schemas/checklist_padrao.py
"""Pydantic schemas for ItemChecklistPadrao entity."""

from datetime import datetime

from pydantic import BaseModel, Field


class ItemChecklistPadraoBase(BaseModel):
    nome_item: str = Field(..., max_length=200)
    descricao: str | None = None
    obrigatorio: bool = True
    ordem_exibicao: int = Field(default=0, ge=0)


class ItemChecklistPadraoCreate(ItemChecklistPadraoBase):
    pass


class ItemChecklistPadraoUpdate(BaseModel):
    nome_item: str | None = Field(None, max_length=200)
    descricao: str | None = None
    obrigatorio: bool | None = None
    ordem_exibicao: int | None = Field(None, ge=0)


class ItemChecklistPadraoResponse(ItemChecklistPadraoBase):
    id: int
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/checklist-padrao.ts
/** TypeScript interfaces for ItemChecklistPadrao entity matching Pydantic schemas. */

export interface ItemChecklistPadraoResponse {
  id: number;
  nome_item: string;
  descricao: string | null;
  obrigatorio: boolean;
  ordem_exibicao: number;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface ItemChecklistPadraoCreate {
  nome_item: string;
  descricao?: string | null;
  obrigatorio?: boolean;
  ordem_exibicao?: number;
}

export interface ItemChecklistPadraoUpdate {
  nome_item?: string | null;
  descricao?: string | null;
  obrigatorio?: boolean | null;
  ordem_exibicao?: number | null;
}
